src/gui/rss/articlelistwidget.cpp

```diff
@@ -55,28 +55,38 @@ void ArticleListWidget::setRSSItem(RSS::Item *rssItem, bool unreadOnly)
 {
     // Clear the list first
     clear();
+    m_rssArticleToListItemMapping.clear();
     if (m_rssItem)
         m_rssItem->disconnect(this);
 
     m_unreadOnly = unreadOnly;
     m_rssItem = rssItem;
-    if (!m_rssItem) return;
-
-    m_rssItem = rssItem;
-    connect(m_rssItem, &RSS::Item::newArticle, this, &ArticleListWidget::handleArticleAdded);
-    connect(m_rssItem, &RSS::Item::articleRead, this, &ArticleListWidget::handleArticleRead);
-    connect(m_rssItem, &RSS::Item::articleAboutToBeRemoved, this, &ArticleListWidget::handleArticleAboutToBeRemoved);
-
-    foreach (auto article, rssItem->articles()) {
-        if (!(m_unreadOnly && article->isRead()))
-            addItem(createItem(article));
+    if (m_rssItem) {
```

Comment:
The only problem is that when `doChecks == false`, compiler detects that `checker` is unused variable, since it has a trivial constructor and destructor...